### ISSUE TITLE
Media Library - Delete via Dropdown Menu from Hover

### ISSUE THREAD

**Issue by jauyong** [GoogleForCreators/web-stories-wp#1319]
## Feature Description

As an author I want to be able to delete media files from my library

Note:
See related issue #354 

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance Criteria

1. While hovering over a media file the more icon is shown.
2. After clicking delete in the dropdown menu, the media file is removed from the library and the files below are shifted up

## QA Instructions

1. hover over a file in the library to trigger the drop down
2. select delete from the dropdown

![hover state](https://user-images.githubusercontent.com/15134432/80561669-8dc45280-8a28-11ea-9289-538a91b9b1bc.png)

**Comment by spacedmonkey** [comment]
As discussed with @joannag6 

Untested code, but a simple delete method would look like this. 
```
 /**
   * Delete Existing media.
   *
   * @param  {number} mediaId
   * @return {Promise} Media Object Promise.
   */
  const deleteMedia = useCallback(
    (mediaId) => {
      return apiFetch({
        path: `${media}/${mediaId}`,
        method: 'DELETE',
      });
    },
    [media]
  );
```

**Comment by spacedmonkey** [comment]
I would add a dialog to confirm deletion of a media item. Thoughts @samitron7

**Comment by o-fernandez** [comment]
@samitron7 we should consider this carefully, in particular, how do we show the appropriate deletion warning to the user. As it currently stands, when the user deletes a media item from their library, it disappears from anywhere that media file may have been used. So, we wouldn't want users in the Story Editor mistakenly delete files that then break image files on their site pages or articles. 

So, two questions: 

1. Do we really need the delete capability in the editor? Users can always manage their uploaded media from the WordPress media tools. 
2. If we must have it in the story editor, can we restrict delete to **only** files uploaded through the tool? 

Let's make sure this is properly thought out before implementing it. For now, adding labels for UX and product consideration.

**Comment by swissspidy** [comment]
> If we must have it in the story editor, can we restrict delete to only files uploaded through the tool?

Technically it's possible, dependent on #342. But would still mean you could potentially delete media files that are used in another story.

**Comment by samitron7** [comment]
Wait...that does not sound right. If I delete an image, I would expect it to delete it from my media library and not all the stories that had that image. Can someone on eng confirm what is going on if you delete an image/video. My assumption has always been, I delete this image, the only place it gets removed is the media library. If this is not the case, then we should try to fix that because it's a pretty bad experience.

**Comment by o-fernandez** [comment]
> Wait...that does not sound right. If I delete an image, I would expect it to delete it from my media library and not all the stories that had that image. Can someone on eng confirm what is going on if you delete an image/video. My assumption has always been, I delete this image, the only place it gets removed is the media library. If this is not the case, then we should try to fix that because it's a pretty bad experience.

To confirm, if you delete an image from the media library in WordPress, the file is removed and all stories that use that image (link to it as part of the story) would no longer show that image as the asset won't load.

**Comment by samitron7** [comment]
For now, we will allow the users to delete media from the editor's media library with the caveat that a dialog box with a warning will pop up each time. [Figma](https://www.figma.com/file/Cd9iacVknHK62RO4pRwb68/Stories-Editor?node-id=7941%3A18814)

**Comment by swissspidy** [comment]
@joannag6 Seems like this one can be closed? What's missing here?

**Comment by joannag6** [comment]
My bad, just added a PR that flips the feature flag on. Will move to Ready for QA.

**Comment by swissspidy** [comment]
PR to test: #2660

**Comment by csossi** [comment]
Verified in QA

**Comment by samitron7** [comment]
Verified in UAT but I filed a new [issue](https://app.zenhub.com/workspaces/web-stories-5e94d3aced449034e1e9b226/issues/google/web-stories-wp/2784) to deal with the following:

1. Add image to page
2. Delete that image in media library

The image on page should be removed and in it's place a broken image symbol. In one instance, I swore I say the broken image symbol hidden in the dark workspace area as I was resizing the browser window but wasn't able to reproduce
